lib/megaLogger.js

```diff
@@ -94,27 +106,19 @@
      * @returns {boolean}
      * @private
      */
-    MegaLogger._environmentHaveSupportForColors = function() {
+    function _environmentHaveSupportForColors() {
         var isBrowser = true;
         var isIE = false;
 
-        if (typeof(window) === 'undefined') {
+        if (typeof(window) === 'undefined' || /PhantomJS/.test(window.navigator.userAgent)) {
             isBrowser = false;
         }
-        else if (typeof(window) !== 'undefined' && /PhantomJS/.test(window.navigator.userAgent)) {
-            isBrowser = false;
-        } else if (
-            typeof(window) !== 'undefined' && (
-            /MSIE/.test(window.navigator.userAgent) ||
-            /Trident/.test(window.navigator.userAgent) ||
-            /Edge/.test(window.navigator.userAgent)
-            )
-        ) {
+        else if ("ActiveXObject" in window) {
```

Comment:
Edge/Spartan is based on Webkit, hence it must support colors. MSIE 11 doesn't identify itself with "msie" in the user agent which is why `Trident` was used here i think, however just checking for `ActiveXObject` should be faster/cleaner than dealing with RegEx and strings since it exists in every MSIE version.